native/Avalonia.Native/src/OSX/WindowImpl.mm

```diff
@@ -366,6 +366,18 @@
     }
 }
 
+HRESULT WindowImpl::GetWindowZOrder(long* zOrder) {
+    START_COM_CALL;
+    @autoreleasepool {
+        if (zOrder == nullptr) {
+            return E_POINTER;
+        }
+
+        *zOrder = [Window orderedIndex];
```

Comment:
This new prop can be used in https://github.com/AvaloniaUI/Avalonia/blob/86bfc26aceeb3000f47e3a67c39daa78deb0acda/samples/IntegrationTestApp/ShowWindowTest.axaml.cs#L78

Reply:
Changed the `GetOrderedIndex` to the new WindowZOrder, I guess those tests could be expanded to all desktop platforms now (though rather than checking the absolute z-order value, it should check its relation to the parent window)